temporalio/worker/worker.py

```diff
@@ -70,10 +75,20 @@ def __init__(
             activities: Set of activity callables decorated with
                 :py:func:`@activity.defn<temporalio.activity.defn>`. Activities
                 may be async functions or non-async functions.
+            workflows: Set of workflow classes decorated with
+                :py:func:`@workflow.defn<temporalio.workflow.defn>`.
             activity_executor: Concurrent executor to use for non-async
                 activities. This is required if any activities are non-async. If
                 this is a :py:class:`concurrent.futures.ProcessPoolExecutor`,
                 all non-async activities must be picklable.
+            workflow_task_executor: Thread pool executor for workflow tasks. If
+                this is not present, a new
+                :py:class:`concurrent.futures.ThreadPoolExecutor` will be
+                created with ``max_workers`` set to
+                ``max_concurrent_workflow_tasks``. The default one will be
```

Comment:
> There shouldn't be any blocking stuff going on in these threads

There is in our case because this is how we do deadlock detection. Essentially, every WFT runs in its own thread. Of course WFTs should only take a couple milliseconds so that is harmless, but technically 5 WFTs running means 5 threads running.

Reply:
Hmm.. do you need a thread per concurrent running workflow? Is it just easier to manage? Why not use a single thread?

Reply:
> do you need a thread per concurrent running workflow?

No. It's a choice.

> Is it just easier to manage? Why not use a single thread?

We could use a single thread, but now you've removed all concurrency from workflow tasks for a worker. The goal of a thread is to detect deadlocks and support concurrent workflow tasks. I could definitely reduce the number here or even use a single thread if we wanted.

Reply:
My main concern is you'd possibly run into problems with consuming too many threads and the associated overhead as is the case with Java which uses the same approach.

Deadlock detection is important, but exists to solve a problem that should be rather temporary. Personally I think it's probably OK to lock up a few other in-flight workflows for a second because of a deadlock in one. The user is going to need to address that immediately in any case. Whereas the thread-per-workflow thing is possibly a problem for the happy path

Reply:
I can understand that. Suggestion on what the default max-threads should be? I can make it something like "5". That sound good? I am checking Java to see how they relate max-concurrent-WFT with thread pool size (though I know we may not want to mimic Java here).

(of course many users might be better off passing in their own thread pools they can share across workers, both activity and workflow)

Reply:
After further discussion, I'll put this at some low number like "4". The GIL prevents most practical concurrency anyways (though newer Python versions are working on improving this).

Reply:
I'm not concerned with thread consumption since we don't actually use a thread per workflow for caching purposes like Java does.
I do tend to agree with @Sushisource though that there's not much benefit to using more than 1 thread here since activations will be processed serially anyways, since there's no I/O in workflow code.
Thread will buy you some concurrency in case the workflow runs CPU intensive work which it technically shouldn't but I don't see any harm in setting this to a very low value like Chad suggests but I wouldn't tie this to the max concurrent WFTs since there's work that can be done concurrently before we call into user code like payload codecs.

Reply:
I have changed this to the CPU count or 4, whichever is higher.